Change scripts to allow .js endings.

Anthony-Fiddes opened the pull request:
It looks like there used to be .ns endings for NS2? So I changed this to
work with the current .js endings. It didn't work for me as it was before. Let me know what you think!

Anthony-Fiddes committed:
Change scripts to allow .js endings.

It looks like there used to be .ns endings for NS2? So I changed this to
work with the current .js endings.

Naliwe commented:
Hey o/ Sorry I wasn't paying attention to that repo lately :)

Yeah we use to have .ns for NetScript, and also JS, but they were pretty much equivalent.
I can see it's not the case anymore!

Thanks for the PR :)

Naliwe commented on bin/buildNS.bash:
```diff
@@ -1,12 +0,0 @@
-
-echo "$jsFiles" | while read -r line; do
-  echo "> Building file ${line%.*}.ns"
-  sed -re 's/(import.*\"\/.*)\"/\1\.ns\"/g' "$line" > "${line%.*}.ns"
```
This was used to make sure the import was in absolute form instead of relative because relative paths did not work. Do they work now? If not, keep this file, just replace ns with js; the script will only make sure path are absolute and not change the file extension :)

Anthony-Fiddes replied:
You're absolutely correct! My bad :)

Anthony-Fiddes replied:
Wait, actually, I am weirdly finding that buildNS.bash is causing my build/ files to be empty... Do you have any idea what could be causing this? I am going to remove it again because I am experiencing issues when it is in. I will keep playing around and let you know what I find

Naliwe replied:
Hmm, what OS are you running on ?

Anthony-Fiddes replied:
PopOS, but I can also try it on WSL Ubuntu

Anthony-Fiddes committed:
Add back buildNS file!

I removed this by accident while I was making my PR. Just changed the
ns's to js's.

Anthony-Fiddes committed:
Add execute permission to generateManifest.bash with only read for all
others

Anthony-Fiddes committed:
Update README.md to reflect the use of .js over .ns

Anthony-Fiddes committed:
Make buildNS.bash executable as well (0544 perms)

Anthony-Fiddes committed:
Add buildNS back to build command

Anthony-Fiddes committed:
Remove buildNS because it is breaking the template

EGrunzke commented:
I made similar changes and was going to open a PR myself. Obviously this one already exists!  So `+1` instead!

Naliwe commented:
Hi o/

Sorry again for the late reply. This PR seems to work for me, does it also work for you guys?

Anthony-Fiddes commented:
Yeah it's been working for me! I've been using the new version of the script in my bitburner repo @ https://github.com/Anthony-Fiddes/bitburner-ts with no noticeable problems.

Naliwe committed:
Merge branch 'master' into master
